File: app/src/main/java/com/annapol04/munchkin/db/DataTypeConverter.java
package com.annapol04.munchkin.db;

import android.arch.persistence.room.TypeConverter;

import com.annapol04.munchkin.engine.DataType;

public class DataTypeConverter {
    @TypeConverter
    public static int toInt(DataType dataType) {
        return dataType.ordinal();
    }

    @TypeConverter
    public static DataType toDataType(int id) {
        return DataType.fromId(id);
    }
}

File: app/src/main/java/com/annapol04/munchkin/db/DataTypeConverter.kt
package com.annapol04.munchkin.db

import android.arch.persistence.room.TypeConverter

import com.annapol04.munchkin.engine.DataType

class DataTypeConverter {
    @TypeConverter
    fun toInt(dataType: DataType): Int {
        return dataType.ordinal
    }

    @TypeConverter
    fun toDataType(id: Int): DataType {
        return DataType.fromId(id)
    }
}
